torch/distributed/_shard/sharded_tensor/api.py

```diff
@@ -611,11 +545,11 @@ def reshard(self, resharding_spec: ShardingSpec) -> ShardedTensor:
             tensor([[3], [3], [5], [5], [7], [7], [9], [9]]) # Rank 2
             tensor([[4], [4], [6], [6], [8], [8], [10], [10]]) # Rank 3
         """
-        if (
-            not isinstance(resharding_spec, ChunkShardingSpec) or
-            not isinstance(self._sharding_spec, ChunkShardingSpec)
-        ):
-            raise NotImplementedError("Only ChunkShardingSpec supported for reshard.")
+        # if (
+        #     not isinstance(resharding_spec, ChunkShardingSpec) or
+        #     not isinstance(self._sharding_spec, ChunkShardingSpec)
+        # ):
+        #     raise NotImplementedError("Only ChunkShardingSpec supported for reshard.")
```

Comment:
Yeah you are right, I just reverted this, I guess we can make this generic, but i'm not sure how hard this would be in terms of resharding with Enumerable or other sharding spec.